src/openzaak/components/zaken/api/serializers/zaken.py

```diff
@@ -802,3 +805,34 @@ class Meta:
     def create(self, validated_data):
         validated_data["zaak"] = self.context["parent_object"]
         return super().create(validated_data)
+
+
+class ZaakContactMomentSerializer(serializers.HyperlinkedModelSerializer):
+    class Meta:
+        model = ZaakContactMoment
+        fields = ("url", "uuid", "zaak", "contactmoment")
+        extra_kwargs = {
+            "url": {"lookup_field": "uuid"},
+            "uuid": {"read_only": True},
+            "zaak": {"lookup_field": "uuid"},
+            "contactmoment": {
+                "validators": [
+                    ResourceValidator(
+                        "ContactMoment", settings.CMC_API_SPEC, get_auth=get_auth
+                    )
+                ]
+            },
+        }
+
+    def save(self, **kwargs):
+        try:
+            return super().save(**kwargs)
+        except SyncError as sync_error:
+            # delete the object again
+            ZaakContactMoment.objects.filter(
+                contactmoment=self.validated_data["contactmoment"],
+                zaak=self.validated_data["zaak"],
+            )._raw_delete("default")
+            raise serializers.ValidationError(
+                {api_settings.NON_FIELD_ERRORS_KEY: sync_error.args[0]}
+            ) from sync_error
```

Comment:
I don't see this code pattern anywhere else in Open Zaak - similar cases should be how we handle ZIO/OIO and BIO/OIO, no?

For ZIO, I see the pattern:

```py
        # manual transaction management - documents API checks that the ZIO
        # exists, so that transaction must be committed.
        # If it fails in any other way, we need to handle that by rolling back
        # the ZIO creation.
        try:
            response = create_remote_oio(io_url, zaak_url)
        except Exception as exception:
            zio.delete()
            raise serializers.ValidationError(
                {
                    "informatieobject": _(
                        "Could not create remote relation: {exception}"
                    ).format(exception=exception)
                },
                code="pending-relations",
            )
        else:
            zio._objectinformatieobject_url = response["url"]
            zio.save()
```